nutanix/resource_nutanix_ndb_profiles.go

```diff
@@ -0,0 +1,1280 @@
+					Schema: map[string]*schema.Schema{
+						"nx_cluster_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"date_created": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"date_modified": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"owner_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"profile_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceNutanixNDBProfileCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	req := &era.ProfileRequest{}
+	softwareProfile := false
+	if name, ok := d.GetOk("name"); ok {
+		req.Name = utils.StringPtr(name.(string))
+	}
+
+	if des, ok := d.GetOk("description"); ok {
+		req.Description = utils.StringPtr(des.(string))
+	}
+
+	if engType, ok := d.GetOk("engine_type"); ok {
+		req.EngineType = utils.StringPtr(engType.(string))
+	}
+
+	if cp, ok := d.GetOk("compute_profile"); ok {
+		req.Properties = buildComputeProfileRequest(cp)
+		// setting some defaults values which are generated at runtime
+		req.Topology = utils.StringPtr("ALL")
+		req.Type = utils.StringPtr("Compute")
+		req.SystemProfile = false
+		req.DBVersion = utils.StringPtr("ALL")
+	}
+
+	if np, ok := d.GetOk("network_profile"); ok {
+		nps := np.([]interface{})
+
+		for _, v := range nps {
+			val := v.(map[string]interface{})
+
+			if tp, ok := val["topology"]; ok {
+				req.Topology = utils.StringPtr(tp.(string))
+
+				// other details
+				req.Type = utils.StringPtr("Network")
+				req.SystemProfile = false
+				req.DBVersion = utils.StringPtr("ALL")
+			}
+
+			if ps, ok := val["postgres_database"]; ok {
+				req.Properties = expandNetworkProfileProperties(ctx, meta, ps.([]interface{}))
+			}
+
+			if cls, ok := val["version_cluster_association"]; ok {
+				clster := cls.([]interface{})
+				out := make([]*era.VersionClusterAssociation, len(clster))
+				for _, v := range clster {
+					val := v.(map[string]interface{})
+
+					if p1, ok1 := val["nx_cluster_id"]; ok1 {
+						out = append(out, &era.VersionClusterAssociation{
+							NxClusterID: utils.StringPtr(p1.(string)),
+						})
+					}
+				}
+				req.VersionClusterAssociation = out
+			}
+		}
+	}
+
+	if db, ok := d.GetOk("database_parameter_profile"); ok {
+		req.Properties = buildDatabaseProfileProperties(db.([]interface{}))
+
+		// setting some defaults values which are generated at runtime
+		req.Topology = utils.StringPtr("ALL")
+		req.Type = utils.StringPtr("Database_Parameter")
+		req.SystemProfile = false
+		req.DBVersion = utils.StringPtr("ALL")
+	}
+
+	if sp, ok := d.GetOk("software_profile"); ok {
+		softwareProfile = true
+		splist := sp.([]interface{})
+
+		for _, v := range splist {
```

Comment:
Why a loop if only one software profile create is only allowed ?

Reply:
Are we handling different versions here ?

Reply:
schema is a list. Given max item 1 .